app/services/word_watcher.rb

```diff
@@ -116,6 +116,23 @@ def self.censor_text(text)
     censor_text_with_regexp(text, regexp)
   end
 
+  def self.apply_to_text(text)
+    if regexp = WordWatcher.word_matcher_regexp(:censor)
+      text = censor_text_with_regexp(text, regexp)
+    end
+
+    [:replace, :link].each do |type|
```

Comment:
Not sure that’s a better version than yours but since I’ve tinkered with it for a while, I’m just sharing it :sweat_smile: 
```ruby
    %i[replace link]
      .flat_map { |type| word_matcher_regexps(type).to_a }
      .reject(&:blank?)
      .reduce(text) do |memo, (word_regexp, replacement)|
        memo.gsub(Regexp.new(word_regexp)) { |match| "#{match[0]}#{replacement}" }
      end
```

Reply:
That makes me thing about an edge case I encountered last week - what if you have one regexp that "covers" another one? Since we're matching the regexp based on the order they're created, it's kinda hard to anticipate what the final result will be. Especially since the UI sort them "alphabetically" 🤔 

Would it be better to apply the regexp from the "largest" to the "smallest" ?

Here's the edge case I encountered: I'm using the Watched Word feature to automatically add a link to product pages whenever a product is mentioned in a post. Last week, the company released a new version of one of their products and they named it by adding a word to the previous version product name. Let's say the old version was "SHOE" and the newer version of the product is "SHOE PREMIUM". Since both versions are being produced and sold we have to linkify both. The regexp I'm using looks like this `\bSHOE?\b` for the original version and `\bSHOE PREMIUM\b` for the newer version. But depending on which regexp I create first, when a post contains "SHOE PREMIUM", the result will be different based on which regexp was applied first.

Reply:
But actually if we linkify `SHOE PREMIUM`, the text will still be there and `SHOE` will still match no? :thinking:

Reply:
@Flink, I like the call-chain, but I think that `.reject(&:blank?)` is no longer necessary.

@ZogStriP, that's a long standing issue. I believe it is bigger than this PR, so I will open a new one. I will have to fix the problem in PrettyText too.

Reply:
Yeah definitely. To be honest, I'm also fine with just respecting the order of creation of the regexps. As long as the display is consistent though.